app/Http/Controllers/Forum/ThreadsController.php

```diff
@@ -91,6 +91,25 @@ public function create(): View
 
     public function store(ThreadRequest $request): RedirectResponse
     {
+        $userId = Auth::id(); // Get the authenticated user's ID
+        $midnight = now()->endOfDay();
+        $remainingSeconds = $midnight->diffInSeconds(now());
+
+        // Count the threads posted by the user today
+        $cacheKey = "user_threads_count_{$userId}";
+        $threadCount = Cache::remember($cacheKey, $remainingSeconds, function () use ($userId) {
+            return Thread::where('author_id', $userId)
+                ->where('created_at', '>=', now()->startOfDay())
+                ->count();
+        });
```

Comment:
I wouldn't cache the results tbh. Maybe we can simply get the count by default with `withCount`: https://laravel.com/docs/11.x/eloquent-relationships#counting-related-models

Reply:
@driesvints  can i use default alert of javascript

Reply:
In an ideal world I'd add a new "warning" method to `SendsAlerts` that gets implemented in the UI as well. But let's use `$this->error(` for now.